Update service_definitions.md (#24067)
DataDog/documentation: content/en/service_catalog/service_definitions.md

- V2.1 supports additional UI elements such as service groupings and fields like `application`, `tier`, and `lifecycle`. `Application`, along with Teams, can be used as grouping variables in Service Catalog. `Lifecycle` helps you differentiate between `production`, `experimental`, or `deprecated` services to indicate development stages and apply different reliability and availability requirements. `Tier` indicates the criticality of services, to prioritize during incident triage. For example, `tier 1` typically represents the most critical services whose failure would result in severe customer impact, whereas `tier 4` services typically have no impacts on actual customer experience.
- V2.2 supports user annotation and overwriting auto-detected service type and languages using the fields `type` and `languages`. It also adds support for associating CI pipelines with a service using the field `ci-pipeline-fingerprints`. This version also includes less restrictive validation logic for `contact.type` and `link.type`, so users should expect fewer warnings while submitting YAML.

> - V3.0 adds a `kind` field that supports schemas for additional component types including system, queues, and datastores. Any components within an `system` implicitly inherit its metadata. Furthermore, this version supports manually declaring dependency relationships, in addition to the auto-detected topology through Distributed Tracing and Universal Service Monitoring. In v3.0, we have replaced `application` with `system`.

Reviewer: ```suggestion
- V3.0 adds a `kind` field that supports schemas for additional component types including systems, queues, and datastores. Any components within a `system` implicitly inherit its metadata. Furthermore, this version supports manually declaring dependency relationships, in addition to the auto-detected topology through distributed tracing and Universal Service Monitoring. In v3.0, `application` is replaced with `system`.
```

For more information about the latest updates, see the [schemas][2] on GitHub.

{{< callout url="https://forms.gle/L5zXVkKr5bAzbdMD9" d_target="#signupModal" btn_hidden="false" header="Opt in to the private beta for metadata schema v3.0!" >}}
{{< /callout >}}

### Metadata Schema v3.0 (beta) 
The Entity Definition Schema is a structure that contains basic information about an entity. See the [full schema on GitHub][1].

{{< callout type="info" header="Change to Field Name in the v3.0 Schema " >}}

> For current beta customers using v3.0, we have replaced `application` with `system` in our documentation to match the updated public schema terminology. 

Reviewer: ```suggestion
For beta customers using v3.0, <code>application</code> has been replaced with <code>system</code> in the documentation to match the updated public schema terminology. 
```

{{< /callout >}}

#### New features in v3.0
##### Expanded data model
v3.0 supports multiple kinds of entities. You can organize your systems using various components such as systems, services, queues, and datastores.

##### Enhanced relationship mapping
With APM and USM data, you can automatically detect dependencies among components. v3.0 supports manual declaration to augment auto-detected system topology to ensure a complete overview of how components interact within your systems.

##### Inheritance of system metadata

> Components within an system automatically inherit the system's metadata. It's no longer necessary to declare metadata for all related components one-by-one as in v2.1 and v2.2. 

Reviewer: ```suggestion
Components within a system automatically inherit the system's metadata. It's no longer necessary to declare metadata for all related components one-by-one as in v2.1 and v2.2. 
```
